### PR TITLE
Fix time zone related tests

### LINKED ISSUE #3794: time zone related test failures
Since the recent isodatetime upgrade in Cylc8 two tests are failing on non-UTC time zones.

* tests/functional/cyclers/20-multidaily_local
* tests/functional/cylc-show/03-clock-triggered-non-utc-mode

We should probably consider running tests in non-UTC time zones to catch any potential issues as part of the fix.

### PR DESCRIPTION
**Pull request by MetRonnie**
(The two tests that expected timezone format to be e.g. +01 instead of +0100)

These changes close #3794

**Requirements check-list**
- [X] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [X] Contains logically grouped changes (else tidy your branch by rebase).
- [X] Does not contain off-topic changes (use other PRs for other changes).
- [X] Does not need tests
- [X] No change log entry required (invisible to users).
- [X] No documentation update required.
- [X] No dependency changes.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -56,6 +56,19 @@ jobs:
...
+            time-zone: 'non-UTC'
+          - python-version: 3.7
+            tests: ['tests/f', '2/4']
+            time-zone: 'non-UTC'
+          - python-version: 3.7
+            tests: ['tests/f', '3/4']
+            time-zone: 'non-UTC'
+          - python-version: 3.7
+            tests: ['tests/f', '4/4']
+            time-zone: 'non-UTC'
```
Comment (oliver-sanders) [inline .github/workflows/test.yml:71]: I don't think there is any need to mess with the matrix, we can just run all functional tests non-UTC.

Comment (MetRonnie) [reply .github/workflows/test.yml]: Oh ok

Comment (MetRonnie) [reply .github/workflows/test.yml]: (I've finished messing around with the matrix now)

**Comment by oliver-sanders** [comment]
tests/f/cylc-graph-diff/00-simple.t   is failing due to #3806, tested locally, passes.

**Review by oliver-sanders — APPROVED** [review]

**Comment by MetRonnie** [comment]
(Just spotted a small mistake)

**Review by hjoliver — APPROVED** [review]

**Pull request merged**